### DISCUSSION TITLE
About Pixiv API Change / 关于 Pixiv 的 API 变更

### DISCUSSION THREAD

**Discussion by y-young** [y-young/nazurin#77] [General]
We've heard that Pixiv no longer support logging in through username & password (upbit/pixivpy#158), but this change **won't bring large effects to this project** as for now.
In detail, if you had successfully used Pixiv functions, Nazurin will cache your `access_token` & `refresh_token` in database, which can be used to log in even without username & password.
In short, please **keep your `refresh_token` safely** and there's nothing to worry about.
If you encountered any related problems, please leave a comment below.

---
我们已经了解到 Pixiv API 不再允许通过用户名和密码登录（upbit/pixivpy#158），但目前来看**此变更不会对本项目造成太大影响**
具体来说，如果曾经成功使用过 Pixiv 相关功能，Nazurin 将在数据库中缓存 `access_token` 和 `refresh_token`，这两种凭据可单独用于登录而无需用户名和密码
因此只需 **妥善保管好数据库中的 `refresh_token`** 即可
如遇到相关问题，请在此 Issue 内留言

**Comment by roytam1** [comment]
and new comers are unable to use now.

**Comment by y-young** [comment]
> and new comers are unable to use now.

Yes, that seems to be the only inconvenience, unless they've already got a `refresh_token`. We'll follow future updates on this change.

**Comment by y-young** [comment]
The author of Pixivpy has released a semi-automatic script to help you obtain a refresh token, see [here](https://gist.github.com/upbit/6edda27cb1644e94183291109b8a5fde).
From Nazurin v1.2, you'll be able to use environment variable `PIXIV_TOKEN` to set refresh token, this option is not required if it is already cached in database.

---
Pixivpy 的作者发布了一个半自动化脚本来辅助获取 Pixiv refresh token，详见[这里](https://gist.github.com/upbit/6edda27cb1644e94183291109b8a5fde)
Nazurin 从 1.2 版本开始支持使用环境变量 `PIXIV_TOKEN` 来设置 refresh token，如果数据库中已经缓存了 refresh token 则无需设置
